### PR TITLE
[STACCMR-47] [STACCMR-48] Add root and default catalogs. Add Logger. Add Settings.

### PR DESCRIPTION
**Pull request by PeteShepley**
Added the root /stac endpoint and the associated logic around that.

Also, moved the current stac search pieces into the API module. Added some tests around new functionality.

### CODE REVIEW

File: search/lib/application.js [hunk 1]
```diff
@@ -2,10 +2,17 @@ const express = require('express');
...
+const settings = require('./settings');
+const { createLogger } = require('./util');
+const logger = createLogger(settings.logger);
+
 const api = require('./api');
 const { errorHandler } = require('./error-handler');
 
 async function initialize () {
+  logger.debug('Initialize Application');
+  logger.silly('This is ridiculous.');
```
Comment (quetcodesfire) [inline search/lib/application.js]: I think we can remove this line as it was just for an example. I would probably remove the line above as well unless you're adding more context.

File: search/lib/api/stac.js [hunk 2]
```diff
@@ -1,7 +1,66 @@
...
+  response.status(200).json(result);
+}
+
+async function postSearch (request, response) {
+  const event = request.apiGateway.event;
+  const result = await search(event, JSON.parse(request.body));
+  response.status(200).json(result);
+}
+
+async function getRootCatalog (request, response) {
```
Comment (quetcodesfire) [inline search/lib/api/stac.js]: Just wondering if this function should be an asynchronous function as it doesn't await anything.

Comment (PeteShepley) [reply search/lib/api/stac.js]: I agree completely. I will remove this.

File: search/lib/application.js (hunk 1 above)
Comment (PeteShepley) [reply search/lib/application.js]: yup, I will remove the silly message.

**Pull request merged**